### PR TITLE
KAFKA-2700: Delete topic should remove the corresponding ACLs

### PR DESCRIPTION
**Pull request by omkreddy**

### CODE REVIEW

**Comment by omkreddy** [comment]
@harshach @junrao we are already deleting  configs and committed offsets associated with a topic. This PR deletes the associated ACLs.  Since we have separate unit tests for topic deletion and acl deletion,  I have not added any specific unit test here.  Pl review.

File: core/src/main/scala/kafka/controller/TopicDeletionManager.scala [hunk 1]
```diff
@@ -73,9 +75,10 @@ import java.util.concurrent.atomic.AtomicBoolean
  * @param initialTopicsToBeDeleted The topics that are queued up for deletion in zookeeper at the time of controller failover
  * @param initialTopicsIneligibleForDeletion The topics ineligible for deletion due to any of the conditions mentioned in #3 above
  */
-class TopicDeletionManager(controller: KafkaController,
-                           initialTopicsToBeDeleted: Set[String] = Set.empty,
-                           initialTopicsIneligibleForDeletion: Set[String] = Set.empty) extends Logging {
+class   TopicDeletionManager(controller: KafkaController,
```
Comment (ijuma) [inline core/src/main/scala/kafka/controller/TopicDeletionManager.scala]: Unnecessary spaces added here.

**Review by ijuma — COMMENTED** [review]
Hmm, do we really want to do this? ACLs have their own lifecycle and they can be added even if a topic doesn't exist right?

So, someone today could create an ACL for a topic, then create/delete/create the topic and the ACL would still be there. This PR changes that.

**Comment by omkreddy** [comment]
@ijuma fair point. But there may be some other scenarios where admins want full control on topic creation and ACLs related to to those topics.  I don't have any strong use case. If others also thinks the same, we can close this JIRA.

**Comment by omkreddy** [comment]
@junrao  Your views on this?  I agree with @ijuma, ACLs can have their own lifecycle.  Users who want to delete ACLs can implement the functionality by using Alter ACL request (KIP-4)

**Comment by omkreddy** [comment]
Closing as per PR discussion.  As of now, this functionality can be implemented by using Java Admin Client.

**Pull request closed without merging**